tic_tac_toe/game.py

```diff
@@ -87,18 +113,55 @@ def move(game, player, position):
     checks before the actual movement is done.
     After registering the movement it must check if the game is over.
     """
-    pass
-
+    current_player = get_next_turn(game)
+    board = game['board']
+    
+    if (game['winner'] != None or _board_is_full(board)):
+        raise InvalidMovement('Game is over.')
+    if (player != current_player): 
+        raise InvalidMovement('"' + current_player + '" moves next.')
+    if not _position_is_valid(position):
+        raise InvalidMovement('Position out of range.')
+    if not _position_is_empty_in_board(position, board):
+        raise InvalidMovement('Position already taken.')
+        
+    board[position[0]][position[1]] = current_player
+    game['winner'] = _check_winning_combinations(board, player)
+    
+    if game['winner']:
+        raise GameOver('"' + current_player + '" wins!')
+    elif _board_is_full(board):
+        raise GameOver('Game is tied!')
+    switch_turn(game)
+    
 
 def get_board_as_string(game):
```

Comment:
This is a little manually exhausting to write and hard to read through to know exactly what you're doing by just looking at the code. What you might've wanted to do is make some ASCII art and use `%s`. Not going to make the whole board but something like below:
```python
return '''
%s | %s | %s|
---------------
''' % tuple([cell for row in board for cell in row])
```